src/audio.c

```diff
@@ -751,20 +779,84 @@ static int stream_pt_handler(uint8_t pt, struct mbuf *mb, void *arg)
 }
 
 
+static int process_decfilt(struct aurx *rx, struct auframe *af)
+{
+	struct le *le;
+
+	/* Process exactly one audio-frame in reverse list order */
+	for (le = rx->filtl.tail; le; le = le->prev) {
+		struct aufilt_dec_st *st = le->data;
+		int err = 0;
+
+		if (st->af && st->af->dech)
+			err = st->af->dech(st, af);
+
+		if (err)
+			return err;
+	}
+
+	return 0;
+}
+
+
+static int rx_push_aubuf(struct aurx *rx, struct auframe *af)
+{
+	int err;
+
+	if (aubuf_cur_size(rx->aubuf) >= rx->aubuf_maxsz) {
+
+		++rx->stats.aubuf_overrun;
+
+#if 0
+		debug("audio: rx aubuf overrun (total %llu)\n",
+		      rx->stats.aubuf_overrun);
+#endif
+	}
+
+	if (rx->auplay_prm.srate != af->srate || rx->auplay_prm.ch != af->ch) {
+		warning("audio: srate/ch of frame %u/%u vs player %u/%u. Use "
+			"module auresamp!\n",
+			af->srate, af->ch,
+			rx->auplay_prm.srate, rx->auplay_prm.ch);
+	}
+
+	if (af->fmt != rx->play_fmt) {
+		warning("audio: rx: invalid sample formats (%s -> %s). %s\n",
+			aufmt_name(af->fmt), aufmt_name(rx->play_fmt),
+			rx->play_fmt == AUFMT_S16LE ? "Use module auconv!" : ""
+		       );
+	}
+
+	err = aubuf_write_auframe(rx->aubuf, af);
+	if (err)
+		return err;
+
+	lock_write_get(rx->lock);
+	if (!rx->aubuf_started &&
+	    (rx->aubuf_minsz >= aubuf_cur_size(rx->aubuf)))
+		rx->aubuf_started = true;
+	lock_rel(rx->lock);
+
+	return 0;
+}
+
+
 static int aurx_stream_decode(struct aurx *rx, const struct rtp_header *hdr,
 			      struct mbuf *mb, unsigned lostc, bool drop)
 {
 	struct auframe af;
 	size_t sampc = AUDIO_SAMPSZ;
 	bool marker = hdr->m;
-	struct le *le;
 	int err = 0;
 	const struct aucodec *ac = rx->ac;
+	bool flush = rx->ssrc != hdr->ssrc && rx->ssrc;
```

Comment:
ssrc value 0 is also a valid SSRC. I think there should be a variable ssrc_set that can be used ?

Reply:
okay, we can simplify this:
```c
bool flush = rx->ssrc != hdr->ssrc;
```
because it doesn't make a difference if we flush at the beginning. The first frame is pushed to `aubuf` anyway.